phantom-jdk8/src/main/scala/com/outworkers/phantom/jdk8/DefaultJava8Primitives.scala

```diff
@@ -37,27 +37,12 @@ trait DefaultJava8Primitives {
     }
   }
 
-  implicit object ZonedDateTimeIsPrimitive extends Primitive[ZonedDateTime] {
+  implicit val zonePrimitive: Primitive[ZoneId] = Primitive.derive[ZoneId, String](_.getId)(ZoneId.of)
 
-    override type PrimitiveType = java.util.Date
-
-    val cassandraType = CQLSyntax.Types.Timestamp
-
-    override def asCql(value: ZonedDateTime): String = {
-      value.toInstant.toEpochMilli.toString
-    }
-
-    override def fromRow(column: String, row: GettableByNameData): Try[ZonedDateTime] = nullCheck(column, row) {
-      r => ZonedDateTime.ofInstant(r.getTimestamp(column).toInstant, ZoneOffset.UTC)
+  implicit val zonedDateTimePrimitive: Primitive[ZonedDateTime] = {
+    Primitive.derive[ZonedDateTime, (Long, String)](dt => dt.toInstant.toEpochMilli -> dt.getZone.getId) {
```

Comment:
I recall datastax driver had similair converter to tuple https://github.com/datastax/java-driver/blob/3.1.x/driver-extras/src/main/java/com/datastax/driver/extras/codecs/jdk8/ZonedDateTimeCodec.java. Generally in this old implementation I intentionally skipped the zone&offset (for offsetdatetime), because with tuple they can't be used for filtering (generally we had 2 columns, one ZonedDateTime and seperarate for Zone), so that any range queries would be possible.